src/scenes/snake-scene.ts

```diff
@@ -38,86 +52,102 @@ export class SnakeScene extends Phaser.Scene {
-    var groundDown = this.add.image(this.tile * 0, this.tile * 16, 'ground').setOrigin(0);
-    var sideLeft = this.add.image(this.tile * -1, this.tile * 0, 'Wallside').setOrigin(0);
-    var sideRight = this.add.image(this.tile * 16, this.tile * 0, 'Wallside').setOrigin(0);
-    Walls.add(groundUp);
-    Walls.add(groundDown);
-    Walls.add(sideLeft);
-    Walls.add(sideRight);
+    this.wallsGroup = this.physics.add.staticGroup();
+    this.wallsGroup.create(this.tile * 0, this.tile * -1, 'ground').setOrigin(0).refreshBody();
+    this.wallsGroup.create(this.tile * 0, this.tile * 16, 'ground').setOrigin(0).refreshBody();
+    this.wallsGroup.create(this.tile * -1, this.tile * 0, 'Wallside').setOrigin(0).refreshBody();
+    this.wallsGroup.create(this.tile * 16, this.tile * 0, 'Wallside').setOrigin(0).refreshBody();
     this.player = this.physics.add.sprite(this.tile, this.tile * 8, 'cube').setOrigin(0);
     this.player.scale = 0.25;
     this.player.setCollideWorldBounds(false);
-    this.physics.add.collider(this.player, Walls, this.collides(this.player, Walls));
+    this.gameOverCollider = this.physics.add.collider(this.player, this.wallsGroup, this.collides(this.player, this.wallsGroup));
     this.player.setBounce(0);
+    this.player.setCollideWorldBounds(false);
     this.cursors = this.input.keyboard.createCursorKeys();
     this.Scores.ShowScore();
+    this.devText = this.add.text(this.tile * 14.53, this.tile * 14.4, this.blinkdev(), { fontSize: '100px', fill: '#00FF00' });
+    this.devText.setText();
+    this.WordText = this.add.text(300, 514, 'Word: ' + this.Nword, { fontSize: '32px', fill: '#000' });
     
     this.timerEvent = this.time.addEvent({
       delay: this.Speed,
       loop: true,
       callback: this.moveSnake,
       callbackScope: this
     });
+    this.Nhashtags = 1;
     
-    this.Ncapsule = 1;
-    const capsule = this.physics.add.sprite(this.tile * 11, this.tile * 8, 'capsule').setOrigin(0);
-    capsule.scale = 0.25;
-    this.physics.add.overlap(this.player, capsule, this.collectCapsule());
+    this.hashtagsGroup = this.physics.add.group();
+    this.physics.add.overlap(this.hashtagsGroup, this.hashtagsGroup, this.collidesHashtag());
+
+    const hashtag = new Hashtags(this, this.tile * 12.5, this.tile * 8.5,'#', '#000');
+    this.physics.add.overlap(this.player, hashtag, this.collectHashtag());
+    this.hashtagsGroup.add(hashtag);
+
   };
 
-  collectCapsule(): ArcadePhysicsCallback {
-    return (player: Phaser.Physics.Arcade.Sprite, capsule: Phaser.Physics.Arcade.Sprite) => {
-      capsule.disableBody(true, true);
+  generateHashtags() {
+    this.Nhashtags = 0;
+    this.word = this.availableHastags[this.getRandomInt(5)]
+
+    for (let letter of this.word) {
+      const hashtag = new Hashtags(this, this.tile * (this.getRandomInt(16) + 0.5), this.tile * (this.getRandomInt(16) + 0.5), letter, this.availableColors[this.getRandomInt(8)]);
```

Comment:
pareil tu peux faire une fonction generateRandomPosition() qui te retourne une position aléatoire this.tile * (this.getRandomInt(16) + 0.5) puisque tu l'utilises à pas mal d'endroits